### PR TITLE
fix for #3750

### LINKED ISSUE #3750: ResultSet miss the plain column  because of using contains to get the logicColumn when using  data mask.
### Which version of ShardingSphere did you use?
sharding-jdbc-4.0.0-RC3
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
sharding-jdbc
### Expected behavior
return the all of table's columns.
### Actual behavior
resultSet miss the logic column  because of using contains to get the logicColumn

### Reason analyze (If you can)
org.apache.shardingsphere.shardingjdbc.jdbc.core.resultset.EncryptResultSetMetaData#getLogicColumn
image

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
rules are:
image

The boss_real_name contains real_name cause the resultSet missing.
I think that it should use equals  insteaded

### PR DESCRIPTION
**Pull request by xiyelife**
Fixes #3750 
Changes proposed in this pull request:
use equals instead of contains to get the plain column.

### CODE REVIEW

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public String getColumnName(final int column) throws SQLException {
     
     private String getLogicColumn(final String actualColumn) throws SQLException {
         for (Entry<String, String> entry : logicAndActualColumns.entrySet()) {
-            if (entry.getValue().contains(actualColumn)) {
+            if (entry.getValue().equals(actualColumn)) {
```
Comment (kimmking) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: you should add a test case for this corner case to make sure that this modification works well as we expect.

Comment (SteNicholas) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: @xiyelife You could add unit test in `EncryptStatementTest`.

Comment (kimmking) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: and look over this modification, you can use map.values().contains instead of for statement.

Comment (xiyelife) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: > and look over this modification, you can use map.values().contains instead of for statement.
Here the code want to get the matched value's key, so must use the for each.

Comment (xiyelife) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: > @xiyelife You could add unit test in `EncryptStatementTest`.

Ok  I will.

Comment (xiyelife) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: > @xiyelife You could add unit test in `EncryptStatementTest`.

OK, I will.

**Comment by tristaZero** [comment]
@xiyelife Hi Thans for your revise, welcome more contributions from you and enjoy it.

**Comment by tristaZero** [comment]
[#3809](https://github.com/apache/incubator-shardingsphere/pull/3809) answears the related issue.
